src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs

```diff
@@ -1000,14 +1000,14 @@ public async Task DoesNotEnforceRequestBodyMinimumDataRateOnUpgradedRequest()
         }
 
         [Fact]
-        public async Task SynchronousReadsAllowedByDefault()
+        public async Task SynchronousReadsDisallowedByDefault()
```

Comment:
This test seems to be doing way more than described.  Why is there a difference between first and second requests? Can we consider adding a simpler test that just checks:

```c#
var ioEx = Assert.Throws<InvalidOperationException>(() => context.Request.Body.Read(new byte[1], 0, 1));
Assert.Equal(CoreStrings.SynchronousReadsDisallowed, ioEx.Message);
```

And change the name of the test to what it is actually testing?

Reply:
The second request opts into sync reads via bodyControlFeature.

Reply:
Does it need the first request to begin with? Can the first request opt just do a sync read on the body?

Reply:
I've split this into two tests.